### ISSUE TITLE
[TreeView] Tree Item do not receive the item defined in `RichTreeViewProps['item']`

### ISSUE THREAD

**Issue by sdancer75** [mui/mui-x#12697]
### Summary

There is no way to get the custom item properties (besides the id, label, disabled, children, etc) for extra customization.

### Examples

```

const ITEMS = [
   {
      id: '1',
      label: 'label1',
      customProperty1: false,
      customProperty2: 'ADD',
},

const CustomTreeItem = forwardRef((props, ref) => {
   
   const { id, itemId, label, disabled, children, ...other } = props;

   const { getRootProps, getContentProps, getIconContainerProps, getLabelProps, getGroupTransitionProps, status } = useTreeItem2({
      id,
      itemId,
      children,
      label,
      disabled,
      rootRef: ref,
   });

   console.log('props', props);

   return (
      <TreeItem2Provider itemId={itemId}>
         <TreeItem2Root {...getRootProps(other)}>
            <CustomTreeItemContent {...getContentProps()}>
               <Box sx={{ flexGrow: 1 }}>
                  <Stack direction="row" alignItems="center">
                     <TreeItem2IconContainer {...getIconContainerProps()}>
                        <TreeItem2Icon status={status} />
                     </TreeItem2IconContainer>

                     <TreeItem2Label {...getLabelProps()} />
                  </Stack>
                  {!children && (
                     <Stack direction="row" justifyContent="flex-start" sx={{ pl: 2, ml: 2, mt: 1, backgroundColor: '#2F3B5F', borderRadius: 10, px: 2, my: 1 }}>
                        All spare parts (xxxxx)
                        <Stack sx={{ ml: 'auto' }}>(folder icon here)</Stack>
                     </Stack>
                  )}
               </Box>
            </CustomTreeItemContent>
            {children && <TreeItem2GroupTransition {...getGroupTransitionProps()} />}
         </TreeItem2Root>
      </TreeItem2Provider>
   );
});


 <RichTreeView
    aria-label="icon expansion"
    sx={{ position: 'relative' }}
    items={ITEMS}
    slots={{ item: CustomTreeItem }}
 />
 

```

### Motivation

To customize each item uniquely

**Search keywords**: RichTreeView, Items

**Comment by flaviendelangle** [comment]
Hi,

You can use the `publicAPI` object returned by `useTreeItem2` which has a `getItem` method

We don't pass the rest of the item directly to `TreeItem` because it would mean that `TreeItem` does not receive the same props when used with `SimpleTreeView` and `RichTreeView`, and we want to keep those as similar as possible at the item level.

This method should clearly be documented, and in general, we need a new page for the `useTreeItem2` hook that describes the APIs.

**Comment by sdancer75** [comment]
Excellent, thanks :-) 

`publicAPI `is not clearly stated in the documentation.

**Comment by flaviendelangle** [comment]
I'll keep this one open to keep track of the doc improvement :+1:

**Comment by noraleonte** [comment]
We decided to add a new section in the docs called `Common features`, and group the following pages inside it
- Tree Item
- Accessibility 
- apiRef (https://github.com/mui/mui-x/issues/13123)

The Tree Item page should cover the use of the two `TreeItem` components, the `useTreeItem2` hook, and the `getItem` method from the `apiRef`. We should have at least 2 examples, one showcasing how to entirely customize a tree item, and a second one to showcase the use of `slots`

**Comment by flaviendelangle** [comment]
Closes in favor of #13161
